src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -55,14 +55,42 @@ static constexpr bool debugFeaturesDefault{ false };
 GlobalAppSettings::GlobalAppSettings() :
     _keymap{ winrt::make_self<KeyMapping>() },
     _keybindingsWarnings{},
-    _unparsedDefaultProfile{},
+    _validDefaultProfile{ false },
     _defaultProfile{},
     _DebugFeaturesEnabled{ debugFeaturesDefault }
 {
     _commands = winrt::single_threaded_map<winrt::hstring, Model::Command>();
     _colorSchemes = winrt::single_threaded_map<winrt::hstring, Model::ColorScheme>();
 }
 
+// Method Description:
+// - Copies any extraneous data from the parent before completing a CreateChild call
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void GlobalAppSettings::_FinalizeInheritance()

```

Comment:
> So, to get this right, every node in the GlobalAppSettings inheritance graph has a full copy of the parent's schemes and commands, correct?

Minor thing I messed up. These should be `std::move` instead of copies. That way we don't hold on to an extra copy of color schemes, commands, etc. that we'll never touch. So instead we'll just have one copy of all of this and pass it down to the leaf.

>  So for these, we can't really tell what layer a command was actually defined at, right? But there's no other way of doing this, because we can't easily express "Layer 2 undefined {command} from layer 1", right?

Exactly. This goes back to the discussion we were having about doing action IDs sooner than later. With action IDs, we should be able to layer an action onto our ancestors. Additionally, we'll have to design our own data structure solution that...
- checks the leaf for an action
- if none are found, checks the parent's
while also maintaining more complex functionality like "unbind" and "rebind" a command. These two work items are pretty intertwined IMO.